### PR TITLE
Avoid calling getElementValue, for efficiency

### PR DESCRIPTION
**Pull request by mernst**
Merge after ~https://github.com/typetools/checker-framework/pull/4290~.

### CODE REVIEW

File: javacutil/src/main/java/org/checkerframework/javacutil/AnnotationUtils.java [hunk 1]
```diff
@@ -684,9 +684,13 @@ public static boolean hasElementValue(AnnotationMirror anno, CharSequence elemen
      * Get the element with the name {@code elementName} of the annotation {@code anno}. The result
      * is expected to have type {@code expectedType}.
      *
-     * <p>For elements of array type, use {@code getElementValueArray} instead.
+     * <p>If the return type is an array, use {@link #getElementValueArray} instead.
      *
-     * <p>For elements of enum type, use {@code getElementValueEnum} instead.
+     * <p>If the return type is an enum, use {@link #getElementValueEnum} instead.
+     *
+     * <p>If possible, it is more efficient to use {@code
```
Comment (smillst) [inline javacutil/src/main/java/org/checkerframework/javacutil/AnnotationUtils.java:691]: Just for my own understanding, this comment is true for all the `getElementValue*` method right?  Also, if the annotation only has one element, is it still more efficient to call `anno.getElementValues().get(someElement).getValue()`?

File: framework/src/main/java/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java [hunk 2]
```diff
@@ -1006,12 +1081,15 @@ public AnnotationMirror convertIntValToIntRange(AnnotationMirror intValAnno) {
...
+     * Returns a {@link Range} bounded by the values specified in the given {@code @Range}
      * annotation. Also returns an appropriate range if an {@code @IntVal} annotation is passed.
      * Returns {@code null} if the annotation is null or if the annotation is not an {@code
      * IntRange}, {@code IntRangeFromPositive}, {@code IntVal}, or {@code ArrayLenRange}.
+     *
+     * @param rangeAnno a {@code @Range} annotation
+     * @return the {@link Range} that the annotation represents
      */
-    public static Range getRange(AnnotationMirror rangeAnno) {
+    public Range getRange(AnnotationMirror rangeAnno) {
```
Comment (smillst) [inline framework/src/main/java/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java:1092]: This method still has calls to `getElementValue`.  Should they be replaced with calls to `getIntRangeToValue`, `getArrayLenRangeToValue`, and other methods?  In general, should all the calls to `getElementValue` in the Value Checker be replace?

File: framework/src/main/java/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java [hunk 3]
```diff
@@ -350,13 +370,68 @@ public long getFromValueFromIntRange(AnnotatedTypeMirror atm) {
...
+        return (long) anno.getElementValues().get(intRangeToElement).getValue();
+    }
+
+    /**
+     * Gets the from() element/field out of an ArrayLenRange annotation.
+     *
+     * @param anno an ArrayLenRange annotation
+     * @return its from() element/field
+     */
+    @SuppressWarnings("UnusedMethod")
```
Comment (smillst) [inline framework/src/main/java/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java:408]: I think this method should be used in getRange.

File: framework/src/main/java/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java [hunk 4]
```diff
@@ -350,13 +370,68 @@ public long getFromValueFromIntRange(AnnotatedTypeMirror atm) {
...
+    }
+
+    /**
+     * Gets the to() element/field out of an ArrayLenRange annotation.
+     *
+     * @param anno an ArrayLenRange annotation
+     * @return its to() element/field
+     */
+    @SuppressWarnings("UnusedMethod")
+    private int getArrayLenRangeToValue(AnnotationMirror anno) {
```
Comment (smillst) [inline framework/src/main/java/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java:420]: I think this method should be used in getRange.

File: framework/src/main/java/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java [hunk 5]
```diff
@@ -350,13 +370,68 @@ public long getFromValueFromIntRange(AnnotatedTypeMirror atm) {
...
+    }
+
+    /**
+     * Gets the value() element/field out of a MinLen annotation.
+     *
+     * @param anno a MinLen annotation
+     * @return its value() element/field
+     */
+    @SuppressWarnings("UnusedMethod")
+    private int getMinLenValueValue(AnnotationMirror anno) {
```
Comment (smillst) [inline framework/src/main/java/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java:431]: This method should be called in getSpecifiedMinLenValue.

**Review by smillst — APPROVED** [review]

File: javacutil/src/main/java/org/checkerframework/javacutil/AnnotationUtils.java (hunk 1 above)
Comment (mernst) [reply javacutil/src/main/java/org/checkerframework/javacutil/AnnotationUtils.java:691]: Yes to both questions.  I updated the comment.

**Pull request merged**